lib/util/transformer/length-based-splitter.js

```diff
@@ -3,13 +3,14 @@ import Stream from 'stream';
 
 class LengthBasedSplitter extends Stream.Transform {
 
-  constructor (littleEndian, maxFrameLength, lengthFieldOffset, lengthFieldLength, lengthAdjustment) {
+  constructor (littleEndian, maxFrameLength, lengthFieldOffset, lengthFieldLength, lengthAdjustment, decodeSingleObject = true) {
```

Comment:
how about passing arguments in a dictionary? this would be more effective and easier to read